sources/Outputs/src/DydVRRemote.cpp

```diff
@@ -0,0 +1,92 @@
+                            const algo::GeneratorDefinitionAlgorithm::BusGenMap& busesRegulatedBySeveralGenerators,
+                            const std::string& basename) {
+  using BusId = std::string;
+  std::set<BusId> regulatedBusIds;
+  for (const auto& busToGenerator : busesRegulatedBySeveralGenerators)
+    regulatedBusIds.insert(busToGenerator.first);
+  for (const auto& busToVSC : hvdcDefinitions_.vscBusVSCDefinitionsMap)
+    regulatedBusIds.insert(busToVSC.first);
+
+  for (const BusId& regulatedBusId : regulatedBusIds) {
+    std::string id = constants::modelSignalNQprefix_ + regulatedBusId;
+    boost::shared_ptr<dynamicdata::BlackBoxModel> blackBoxModelVRRemote = helper::buildBlackBox(id, "VRRemote", basename + ".par", id);
+    dynamicModelsToConnect->addModel(blackBoxModelVRRemote);
+    dynamicModelsToConnect->addConnect(id, "vrremote_URegulated", constants::networkModelName, regulatedBusId + "_U_value");
+  }
+
+  writeMacroConnector(dynamicModelsToConnect);
+  writeMacroConnect(dynamicModelsToConnect);
+  writeConnect(dynamicModelsToConnect);
+}
+
+void
+DydVRRemote::writeMacroConnector(boost::shared_ptr<dynamicdata::DynamicModelsCollection>& dynamicModelsToConnect) {
+  if (!generatorDefinitions_.empty()) {
+    auto connector = dynamicdata::MacroConnectorFactory::newMacroConnector(macroConnectorGenVRRemoteName_);
+    connector->addConnect("generator_NQ", "vrremote_NQ");
+    connector->addConnect("generator_limUQUp", "vrremote_limUQUp_@INDEX@_");
+    connector->addConnect("generator_limUQDown", "vrremote_limUQDown_@INDEX@_");
+    dynamicModelsToConnect->addMacroConnector(connector);
+  }
+}
+
+void
+DydVRRemote::writeMacroConnect(boost::shared_ptr<dynamicdata::DynamicModelsCollection>& dynamicModelsToConnect) {
+  std::unordered_map<std::string, unsigned int> modelNQIdGenNumber;
+  for (auto it = generatorDefinitions_.cbegin(); it != generatorDefinitions_.cend(); ++it) {
+    if (it->isNetwork()) {
+      continue;
+    }
+    if (it->model == algo::GeneratorDefinition::ModelType::PROP_SIGNALN_INFINITE ||
+               it->model == algo::GeneratorDefinition::ModelType::PROP_DIAGRAM_PQ_SIGNALN ||
+               it->model == algo::GeneratorDefinition::ModelType::PROP_SIGNALN_RECTANGULAR) {
+      std::string modelNQId = constants::modelSignalNQprefix_ + it->regulatedBusId;
+      ++modelNQIdGenNumber[modelNQId];
+      auto connection = dynamicdata::MacroConnectFactory::newMacroConnect(macroConnectorGenVRRemoteName_, it->id, modelNQId);
+      connection->setIndex2(std::to_string(modelNQIdGenNumber[modelNQId]));
+      dynamicModelsToConnect->addMacroConnect(connection);
+    }
+  }
+}
+
+void
+DydVRRemote::writeConnect(boost::shared_ptr<dynamicdata::DynamicModelsCollection>& dynamicModelsToConnect) {
+  const std::string vrremoteNqValue("vrremote_NQ");
+  for (const auto& keyValue : hvdcDefinitions_.hvdcLines) {
+    algo::HVDCDefinition hvdcLine = keyValue.second;
```

Comment:
why 2 methods to create connections?

Reply:
Currently, the connections creation is splited in 3 methods : 

writeMacroConnector(dynamicModelsToConnect);
writeMacroConnect(dynamicModelsToConnect);
writeConnect(dynamicModelsToConnect);

but I can merge them into one big method : writeAllConnections

Reply:
writeMacroConnector can be keep as it creates the macros, however the two others can be merged IMHO